lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -2610,25 +2622,33 @@ def __init__(self, client):
             self.results['ansible_facts'] = {'docker_container': self.facts}
             self.results['container'] = self.facts
 
-    def wait_for_state(self, container_id, complete_states=None, wait_states=None, accept_removal=False):
+    def wait_for_state(self, container_id, complete_states=None, wait_states=None, accept_removal=False, max_wait=None):
         delay = 1.0
```

Comment:
```suggestion
        if max_wait is not None:
            delay = max_wait:
        else:
            delay = 1.0
```
or something? just the default delay is 1.0 and you can set it to < or > of this value.
The added code from line 2644 looks a bit complex to me

Reply:
That would not be correct, since maximal wait is not equal to the total wait. The module should wait at most `max_wait`, but not in one go, but do smaller waits (if appropriate) until the maximal wait time is reached.

Reply:
The code in lines 2644 to 2649 makes sure that the total wait time never exceeds `max_wait`.

Reply:
ok, thanks for the explanation:) just wanted to be sure you did exactly what you intended to do